### PR TITLE
Feature/topic-service

### PR DESCRIPTION
**Pull request by mwtz**
## Links:

[Related ticket](https://trello.com/c/asHv9Db3/4-3-as-a-user-i-should-be-able-to-create-a-new-target-so-i-can-find-people-with-similar-interests-in-a-specific-area)

## What & Why:

Service for Topics of target 

image

## Risks, Mitigation & Rollback:

- [ ] Safe to Revert *check this box if this PR can be reverted without incident*

### CODE REVIEW

File: src/components/form/select/index.js [hunk 1]
```diff
@@ -9,11 +9,14 @@ const Select = ({ register, options, name, error, placeholder, className = '', .
       <option value="" disabled selected className="placeholder">
         {placeholder}
       </option>
-      {options.map((value, i) => (
-        <option key={i} value={value}>
-          {value.label ? value.label : value}
-        </option>
-      ))}
+      {options.map((value, i) => {
+        console.log(value);
```
Comment (diebarral) [inline src/components/form/select/index.js]: Remove log

File: src/components/form/select/index.js [hunk 2]
```diff
@@ -9,11 +9,14 @@ const Select = ({ register, options, name, error, placeholder, className = '', .
...
+        return (
+          <option key={i} value={value.topic ? value.topic.id : value}>
+            {value.topic ? value.topic.label : value}
+          </option>
```
Comment (diebarral) [inline src/components/form/select/index.js]: I don't understand a thing. If there's no `value.topic`, then what's `value`? Why do you set it as an option?

File: src/components/form/select/index.js (hunk 1 above)
Comment (asdolo) [inline src/components/form/select/index.js]: You can remove this `console.log`.

File: src/components/form/select/index.js (hunk 2 above)
Comment (asdolo) [inline src/components/form/select/index.js]: The `Select` component defined in this file should be generic. It shouldn't check for a `topic` object in the `options` prop. Instead, you should pass a proper `options` prop for the `Select` component at `src/pages/createTarget/index.js`.

All you should do is to pass an array of objects to the `options` prop that contains a `label` property for showing it to the user.

File: src/pages/createTarget/index.js [hunk 3]
```diff
@@ -56,7 +60,7 @@ const CreateTarget = () => {
           <label htmlFor="topic">{t('newTarget.labels.topic')}</label>
           <Select
             name="topic"
-            options={[...TARGET_TOPICS]}
+            options={[...topics]}
```
Comment (asdolo) [inline src/pages/createTarget/index.js]: Instead of modifying the `Select` component (which was ok before this, no changes needed), you should map your `topics` array here to match the interface needed for the `options` prop of the `Select` component.

```js
option={topics.map(topic => topic.topic)}
```

or:

```js
option={topics.map(({ topic }) => topic)}
```

**Review by asdolo — CHANGES_REQUESTED** [review]

File: src/components/form/select/index.js [hunk 4]
```diff
@@ -10,8 +10,8 @@ const Select = ({ register, options, name, error, placeholder, className = '', .
         {placeholder}
       </option>
       {options.map((value, i) => (
-        <option key={i} value={value}>
-          {value.label ? value.label : value}
+        <option key={i} value={value.topic ? value.topic.id : value}>
```
Comment (asdolo) [inline src/components/form/select/index.js]: The `Select` component defined in this file should be generic. It shouldn't check for a `topic` object in the `options` prop. Instead, you should pass a proper `options` prop for the `Select` component at `src/pages/createTarget/index.js`.

I would revert this change. Check my comment above.

**Review by asdolo — CHANGES_REQUESTED** [review]

**Review by asdolo — APPROVED** [review]
Great! 🚀 🙌

**Review by diebarral — APPROVED** [review]
